src/Kdyby/Doctrine/DI/OrmExtension.php

```diff
@@ -382,23 +382,24 @@ protected function processEntityManager($name, array $defaults)
 			$entityManager->addSetup('?->bindEntityManager(?)', [$this->prefix('@' . $name . '.diagnosticsPanel'), '@self']);
 		}
 
-		$builder->addDefinition($this->prefix('repositoryFactory.' . $name . '.defaultRepositoryFactory'))
-				->setClass($config['defaultRepositoryClassName'])
+		$builder->addFactoryDefinition($this->prefix('repositoryFactory.' . $name . '.defaultRepositoryFactory'))
 				->setImplement(IRepositoryFactory::class)
-				->setArguments([new Code\PhpLiteral('$entityManager'), new Code\PhpLiteral('$classMetadata')])
 				->setParameters([EntityManagerInterface::class . ' entityManager', Doctrine\ORM\Mapping\ClassMetadata::class . ' classMetadata'])
-				->setAutowired(FALSE);
+				->getResultDefinition()
+					->setFactory($config['defaultRepositoryClassName'])
+					->setArguments([new Code\PhpLiteral('$entityManager'), new Code\PhpLiteral('$classMetadata')])
+					->setAutowired(FALSE);
```

Comment:
wrong indentation

Reply:
I was following the example of Nette itself, as the last three lines are all the methods of object returned by getResultDefinition().
See: https://github.com/nette/application/blob/master/src/Bridges/ApplicationDI/LatteExtension.php#L57:L62

If you find it still unacceptable, I will change all lines to the same indentation.

Reply:
hmm... yeah I'd prefer same indentation, thanks

Reply:
fixed.